### PR TITLE
Kubespray v2.18.0 and containerd runtime

### PR DESCRIPTION
**Pull request by dholt**
Upgrades Kubespray to v2.18.0 and sets containerd as the default k8s runtime instead of docker.

Using containerd also means we're now using the GPU operator by default

### CODE REVIEW

File: playbooks/k8s-cluster.yml [hunk 1]
```diff
@@ -133,7 +153,9 @@
 - include: container/nvidia-docker.yml
   tags:
     - nvidia
-  when: deepops_gpu_operator_enabled | default('false') | bool == false
+  when:
+    - deepops_gpu_operator_enabled | default('false') | bool == false
```
Comment (supertetelman) [inline playbooks/k8s-cluster.yml:157]: Can you change this to use the same "is defined and == false" syntax?

**Comment by supertetelman** [comment]
Before merging this and making GPU Operator the default we should really fix monitoring: https://github.com/NVIDIA/deepops/issues/890

I have an open PR to bump the version and fix DGX support: https://github.com/NVIDIA/deepops/issues/1069

File: playbooks/k8s-cluster.yml [hunk 2]
```diff
@@ -1,6 +1,26 @@
...
 # Kubernetes Cluster Playbook
 
+# Set facts depending on container runtime
+# Use GPU operator when container runtime is not docker
+# etcd_deployment_type must be `host` when container_manager is not `docker`
+- hosts: all
+  tasks:
+    - name: Set facts when not using docker container runtime (default)
+      set_fact:
+        deepops_gpu_operator_enabled: true
```
Comment (supertetelman) [inline playbooks/k8s-cluster.yml:11]: Why are we enforcing GPU Operator with containerd?

File: roles/nvidia-gpu-operator/defaults/main.yml [hunk 3]
```diff
@@ -18,7 +18,7 @@ gpu_operator_namespace: "gpu-operator-resources"
 gpu_operator_grid_config_dir: "{{ deepops_dir }}/gpu_operator"
 
 # Defaults from https://github.com/NVIDIA/gpu-operator/blob/master/deployments/gpu-operator/values.yaml
-gpu_operator_default_runtime: "docker"
+gpu_operator_default_runtime: "containerd"
```
Comment (supertetelman) [inline roles/nvidia-gpu-operator/defaults/main.yml:21]: I'm removing this in a new PR

File: playbooks/k8s-cluster.yml (hunk 2 above)
Comment (dholt) [reply playbooks/k8s-cluster.yml:11]: I don't recall exactly, but I thought it was required

**Comment by supertetelman** [comment]
Before merging this through and essentially making GPU Operator the default path, I propose we get GPU Operator/ DGX support working (https://github.com/NVIDIA/deepops/pull/1074) get DCGM/Prometheus support integrated (https://github.com/NVIDIA/deepops/issues/890)

Until then, could we just check this PR, but keep the default as Docker?

**Comment by supertetelman** [comment]
DGX Support with GPU Operator is complete. GPU Operator integration into our monitoring stack is complete. Just need to merge https://github.com/NVIDIA/deepops/pull/1074 with the content of this PR.

**Comment by ajdecon** [comment]
DeepOps is currently pinned to Kubespray v2.17.1, which is newer than this PR is using.

Additionally, we're seeing issues in DGX OS 5.1.x (#1110) up to Kubespray v2.17.1.

So when we're ready for this PR to move forward, we should bump the Kubespray version to v2.18.0 (or whatever is latest).

**Review by supertetelman — APPROVED** [review]
LGTM.

I'd like to add some additional documentation here eventually and some additional testing to outline the different K8s configurations options and validate them.

I think we have

* Device Plugin + Docker (with nvidia-driver)
* Device Plugin + containerd (with nvidia-driver)
* GPU Operator + Docker (with driver container or with nvidia-driver)
* GPU Operator + containerd(with driver container or with nvidia-driver)

Is Device Plugin with Docker a supported path now? Do we have 6 total core configurations here?

**Pull request merged**